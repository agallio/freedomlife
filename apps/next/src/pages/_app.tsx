import { NextPage } from 'next'
import Head from 'next/head'
import Script from 'next/script'
import { createPortal } from 'react-dom'
import { Toaster } from 'react-hot-toast'
import {
  QueryCache,
  QueryClient,
  QueryClientProvider,
} from '@tanstack/react-query'
// import { Inter } from '@next/font/google'

// Components
import SEO from '~/components/SEO'
import BottomTabBar from '~/components/BottomTabBar'

// Contexts
import { GuideProvider } from '~/contexts/GuideContext'
import { BibleProvider } from '~/contexts/BibleContext'

// Types
import type { AppProps } from 'next/app'

// Styles
import '../styles/globals.css'
import { BibleLocalDatabaseProvider } from '~/contexts/BibleLocalDatabaseContext'
import { useEffect, useState } from 'react'

// react-modal-sheet not yet compatible with @next/font
// commented this code so if it compatibles, enable it again.
// const inter = Inter({ subsets: ['latin'], variable: '--font-sans' })

const queryClient = new QueryClient({
  queryCache: new QueryCache({
    onError: (error, query) => {
      if (query.meta) {
        console.log(query.meta?.errorMessage, error)
        return
      }

      console.log(error)
    },
  }),
})

const MyApp: NextPage<AppProps> = ({ Component, pageProps, router }) => {
  return (
    <>
      <Head>
        <meta name="title" content="freedomlife â€” Alkitab &amp; Panduan Baca" />
      </Head>
      {process.env.NODE_ENV === 'production' && (
        <Script
          async
          defer
          data-website-id="2eee9168-15ac-46de-814a-35725fc77c53"
          src="https://analytics.agallio.xyz/script.js"
        />
      )}
      <SEO />

      {/* <div className={`${inter.variable} font-sans`}> */}
      <QueryClientProvider client={queryClient}>
        <GuideProvider>
          <BibleProvider>
            <BibleLocalDatabaseProvider>
              <ToasterContainer />
              <Component {...pageProps} />

              {router.pathname !== '/404' &&
                router.pathname !== '/persembahan' &&
                router.pathname !== '/learn' && <BottomTabBar />}
            </BibleLocalDatabaseProvider>
          </BibleProvider>
        </GuideProvider>
      </QueryClientProvider>
      {/* </div> */}
    </>
  )
}

function ToasterContainer() {
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    setMounted(true)
  }, [])

  return mounted ? createPortal(<Toaster />, document.body) : undefined
}

export default MyApp
