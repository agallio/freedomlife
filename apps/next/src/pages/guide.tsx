import { useEffect, useState } from 'react'
import { NextPage } from 'next'
import Router from 'next/router'
import Head from 'next/head'
import dynamic from 'next/dynamic'
import { NextSeo } from 'next-seo'

// Components
import JumboHeader from '~/components/JumboHeader'
import PageTransition from '~/components/PageTransition'
import GuideLoading from '~/components/Guide/GuideLoading'
import GuideItem from '~/components/Guide/GuideItem'
const GuideError = dynamic(() => import('~/components/Guide/GuideError'))
const GuideMonthDialog = dynamic(
  () => import('~/components/Guide/GuideMonthDialog'),
)

// Icon Components
import ChevronDownIcon from '~/components/Icons/ChevronDownIcon'

// Context
import { useGuide } from '~/contexts/GuideContext'

// Utils
import { useGuides } from '~/utils/hooks/useFetchedGuide'
import { monthList } from '~/utils/constants'
import dayjs from '~/utils/dayjs'

const Guide: NextPage = () => {
  const [openMonthDialog, setOpenMonthDialog] = useState(false)
  const [monthNumber, setMonthNumber] = useState(dayjs().format('MM'))

  const { setGuideDate } = useGuide()

  const { data, isLoading, isError, refetch } = useGuides({
    month: monthNumber,
  })

  const toBibleWithDate = (date: string) => {
    setGuideDate(date)
    Router.push('/read')
    localStorage.setItem('in_guide', 'true')
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  useEffect(() => {
    refetch()
  }, [monthNumber])

  return (
    <>
      <Head>
        <title>Panduan Baca — freedomlife</title>
      </Head>
      <NextSeo
        title="Panduan Baca — freedomlife"
        description="Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi."
        openGraph={{
          url: 'https://freedomlife.id/guide',
          title: 'Panduan Baca — freedomlife',
          description:
            'Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi.',
          site_name: 'freedomlife',
          images: [
            {
              url: 'https://freedomlife.id/images/og-guide.png',
              alt: `Tulisan dan logo 'freedomlife' disertai dengan keterangan: 'Panduan Baca Bulan Ini'`,
            },
          ],
        }}
        twitter={{
          cardType: 'summary_large_image',
        }}
      />

      <div className="mx-auto mb-20 max-w-sm p-6 sm:max-w-md sm:py-6">
        <JumboHeader />

        <button
          className="mt-2 flex w-full cursor-pointer items-center justify-between rounded-lg border bg-white px-4 py-2 shadow-md focus:ring-1 focus:ring-gray-400 dark:border-0 dark:bg-gray-600 lg:focus:ring-0"
          onClick={() => {
            document.body.style.overflow = 'hidden'
            setOpenMonthDialog(true)
          }}
        >
          <span>
            {monthList.find((item) => item.value === monthNumber)?.name || ''}
          </span>
          <ChevronDownIcon className="w-6 text-gray-500 dark:bg-gray-600 dark:text-white sm:hover:text-opacity-50" />
        </button>

        <PageTransition>
          {isError ? (
            <GuideError monthNumber={monthNumber} />
          ) : !data || isLoading ? (
            <GuideLoading />
          ) : (
            data.map((item, index) => (
              <GuideItem
                key={index}
                item={item}
                index={index}
                toBibleWithDate={toBibleWithDate}
              />
            ))
          )}
        </PageTransition>

        <GuideMonthDialog
          open={openMonthDialog}
          monthNumber={monthNumber}
          setMonthNumber={setMonthNumber}
          onClose={() => {
            document.body.style.overflow = 'visible'
            setOpenMonthDialog(false)
          }}
        />
      </div>
    </>
  )
}

export default Guide
