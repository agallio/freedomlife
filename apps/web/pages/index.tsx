// Screen
import HomeScreen from '@repo/app/features/home'

// Icon Components
import FreedomLifeIcon from '@repo/app/components/icons/freedomlife-icon'

export default function HomePage() {
  return (
    <div className="mx-auto flex w-full max-w-sm flex-col px-6 pb-28 pt-4 sm:max-w-md">
      <div className="flex pb-4">
        <FreedomLifeIcon className="w-[230px]" />
      </div>

      <HomeScreen />
    </div>
  )
}
