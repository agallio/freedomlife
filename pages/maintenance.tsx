import { useEffect, useState } from 'react'
import { Divider, Typography } from '@material-ui/core'

export const Maintenance: React.FC = () => {
  const [darkMode, setDarkMode] = useState(false)

  useEffect(() => {
    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)')
    setDarkMode(prefersDark.matches)
  }, [])

  return (
    <div className="maintenance">
      <img
        src={`icon-${darkMode ? 'dark' : '512x512'}.png`}
        alt="logo"
        className="maintenance__logo"
      />
      <Divider orientation="vertical" className="maintenance__divider" />
      <Typography className="maintenance__text">
        Situs Sedang Dalam Perbaikan
      </Typography>
    </div>
  )
}

export default Maintenance
