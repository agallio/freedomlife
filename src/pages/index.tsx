import { useEffect, useState } from 'react'
import Router from 'next/router'
import { AnimatePresence } from 'framer-motion'

import JumboHeader from '@/components/JumboHeader'
import HomeBox from '@/components/Home/HomeBox'
import NewYearDialog from '@/components/Home/NewYearDialog'
import NewUserBox from '@/components/Home/NewUserBox'

import { useDispatchGuide } from '../store'

import dayjs from '@/utils/dayjs'
import useRequest from '@/utils/hooks/useRequest'

import type { ApiResponse, GuideDataResponse } from '@/types/api'

const Home = (): JSX.Element => {
  const guideDispatch = useDispatchGuide()
  const [open, setOpen] = useState(false)

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: `/api/guide/${dayjs().format('DD-MM-YYYY')}`,
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    guideDispatch({ type: 'SET_GUIDE_PASSAGE', data: '' })
    Router.push('/read?guide=true')
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  useEffect(() => {
    if (open) {
      document.body.style.overflow = 'hidden'
    } else {
      document.body.style.overflow = 'unset'
    }
  }, [open])

  return (
    <div className="max-w-md p-6 mx-auto mb-20 sm:py-6 md:mb-16 landscape:mx-auto">
      <JumboHeader isHome />

      <main>
        <HomeBox data={data} toBible={toBible} />

        <NewUserBox />

        <AnimatePresence>
          {open && <NewYearDialog handleClose={() => setOpen(false)} />}
        </AnimatePresence>
      </main>
    </div>
  )
}

export default Home
