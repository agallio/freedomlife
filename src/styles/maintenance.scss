.maintenance {
  width: 100%;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  @include element('logo') {
    width: 5%;

    @media (max-width: 374px) {
      width: 15%;
    }

    @media (min-width: 375px) and (max-width: 425px) {
      width: 22%;
    }

    @media (prefers-color-scheme: dark) and (max-width: 425px) {
      width: 15%;
    }
  }

  @include element('divider') {
    height: 70px !important;

    @media (max-width: 374px) {
      height: 60px !important;
    }

    @media (min-width: 375px) and (max-width: 425px) {
      height: 65px !important;
    }
  }

  @include element('text') {
    margin-left: 20px !important;

    @media (max-width: 374px) {
      margin-left: 10px !important;
    }

    @media (min-width: 375px) and (max-width: 425px) {
      margin-left: 15px !important;
    }

    @media (prefers-color-scheme: dark) {
      color: #ffffff !important;
    }
  }
}
