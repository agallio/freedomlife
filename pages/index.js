import React, { Component } from 'react';
import { connect } from 'react-redux';
import Router from 'next/router';
import {
  LinearProgress,
  Card,
  CardContent,
  Typography,
  Grid,
  Fab
} from '@material-ui/core';
import moment from 'moment';
import 'moment/locale/id';

import { fetchGuideToday } from '../src/actions/guide';
import { fetchTodayChapter } from '../src/actions/bible';

class Index extends Component {
  componentDidMount = async () => {
    if (this.props.guide.guideToday.date !== moment().format('DD-MM-YYYY')) {
      this.props.fetchGuideToday();
    }
  };

  toBible = () => {
    if (
      this.props.guide.guideToday.date === moment().format('DD-MM-YYYY') &&
      this.props.bible.chapters.passage.length === 0
    ) {
      this.props.fetchTodayChapter();
    }
    Router.push('/bible');
  };

  render() {
    const { isFetching, guideToday } = this.props.guide;

    return (
      <div>
        {isFetching && <LinearProgress color="secondary" />}
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2 className="header-title" style={{ marginTop: 70 }}>
            Freedom Life
          </h2>
          <p className="header-subtitle">
            Aplikasi panduan baca Alkitab setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="styled-card">
              <CardContent>
                <Typography className="bold-text primary" variant="h5">
                  Panduan Hari Ini
                </Typography>
                <Typography className="light-text primary" variant="subtitle1">
                  {moment().format('dddd, LL')}
                </Typography>

                {['PL', 'PB1', 'PB2'].map(item => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="guide-passage-box">
                        <h5 className="guide-passage-box-text">{item}</h5>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {isFetching ? (
                        <LinearProgress />
                      ) : (
                        <Typography
                          className="bold-text primary"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? guideToday.pl_name
                            : item === 'PB1'
                            ? guideToday.pb1_name
                            : item === 'PB2'
                            ? guideToday.pb2_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="light-text primary"
                        variant="subtitle1"
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB1'
                          ? 'Perjanjian Baru 1'
                          : item === 'PB2'
                          ? 'Perjanjian Baru 2'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="guide-passage-box-fab"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={this.toBible}
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    guide: state.guide,
    bible: state.bible
  };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchGuideToday: () => dispatch(fetchGuideToday()),
    fetchTodayChapter: version => dispatch(fetchTodayChapter(version))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Index);
