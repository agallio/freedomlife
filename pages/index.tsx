import Router from 'next/router'
import {
  Card,
  CardContent,
  Fab,
  Fade,
  Grid,
  LinearProgress,
  Typography,
} from '@material-ui/core'
import { Skeleton } from '@material-ui/lab'

import { dayjs, useRequest } from '../src/utils'
import { useDispatchGuide } from '../src/store'

// Types
import type { ApiResponse, GuideDataResponse } from '../src/types'

export const Home: React.FC = () => {
  const guideDispatch = useDispatchGuide()

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: '/api/guide/today',
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    Router.push('/bible')
  }

  return (
    <>
      {!data && <LinearProgress color="secondary" />}
      <Fade in>
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2
            className="header__title"
            data-testid="headerTitle"
            style={{ marginTop: 65 }}
          >
            FreedomLife
          </h2>
          <p className="header__subtitle" data-testid="headerSubtitle">
            Aplikasi Panduan Baca Alkitab Setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="homecard">
              <CardContent>
                <Typography
                  className="homecard__title"
                  data-testid="homecardTitle"
                  style={{ fontSize: '1.5rem !important', lineHeight: 1 }}
                >
                  Panduan Hari Ini
                </Typography>
                <Typography
                  className="homecard__subtitle"
                  data-testid="homecardSubtitle"
                  variant="subtitle1"
                >
                  {dayjs().format('dddd, DD MMMM YYYY')}
                </Typography>

                {['PL', 'PB', 'ALT'].map((item) => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="homecard__pbox">
                        <p
                          className="homecard__pbox__text"
                          data-testid={`homecardPBox_${item}`}
                        >
                          {item}
                        </p>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {!data ? (
                        <Skeleton variant="text" animation="wave" height={25} />
                      ) : (
                        <Typography
                          className="homecard__title"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? data.data?.pl_name
                            : item === 'PB'
                            ? data.data?.pb_name
                            : item === 'ALT'
                            ? data.data?.alt_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="homecard__subtitle"
                        variant="subtitle1"
                        data-testid={`homecardPSub_${item}`}
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB'
                          ? 'Perjanjian Baru'
                          : item === 'ALT'
                          ? 'Tambahan'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="homecard__button"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={toBible}
                  disabled={!data}
                  data-testid="homecardButton"
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </Fade>
    </>
  )
}

export default Home
