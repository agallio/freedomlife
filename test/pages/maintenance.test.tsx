import { render } from '../testUtils'
import { Maintenance } from '../../pages/maintenance'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

describe('Maintenance Page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Maintenance />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('text value is correct', () => {
    const { getByTestId } = render(<Maintenance />, {})
    const maintenanceText = getByTestId('maintenanceText')
    expect(maintenanceText.textContent).toBe('Situs Sedang Dalam Perbaikan')
  })
})
