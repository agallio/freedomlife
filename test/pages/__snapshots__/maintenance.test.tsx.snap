// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintenance Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    __source="[object Object]"
    class="maintenance"
  >
    <img
      __source="[object Object]"
      alt="logo"
      class="maintenance__logo"
      src="icon-512x512.png"
    />
    <hr
      __source="[object Object]"
      class="MuiDivider-root maintenance__divider MuiDivider-vertical"
    />
    <p
      __source="[object Object]"
      class="MuiTypography-root maintenance__text MuiTypography-body1"
      data-testid="maintenanceText"
    >
      Situs Sedang Dalam Perbaikan
    </p>
  </div>
</DocumentFragment>
`;
